### ISSUE TITLE
use wcjs-prebuilt?

### ISSUE THREAD

**Issue by jaruba** [Magics-Group/wcjs-renderer#6]
I made a branch that uses `wcjs-prebuilt`, should we merge this to master?

It can also be used alternatively by linking to the prebuilt branch instead of the master branch from `package.json`.

@Ivshti is [this](https://github.com/RSATom/wcjs-renderer/commit/e20c4bef3f50dffeae02d35171a00d636817eb08) the correct way of implementing `wcjs-prebuilt`?

@RSATom can you think of any reason why it shouldn't be merged?

@luigiplr are you eating pizza right now?

**Comment by RSATom** [comment]
I just afraid I could not keep `.node` files up-to-date, since for example Electron releases out too frequently.

**Comment by RSATom** [comment]
Maybe the best solution could be merge `wcjs-prebuilt` to main repo, and use regular build as failback

**Comment by RSATom** [comment]
@Ivshti, @luigiplr how much work such merge could require?

**Comment by Ivshti** [comment]
@RSATom I think not a lot

But it's okay for electron, because we only have to re-build when chromium major version is bumped, which means every 3-4 (or even more) major electron releases. E.g. there's need for recompile between 0.29 and 0.30, because it changes chromium's major, but no need to re-compile between 0.26 - 0.30 because they all have the same chromium major.

@RSATom - I think your concern is not that relevant.

**Comment by RSATom** [comment]
@Ivshti, But do `wcjs-prebuilt` use only major version on binary download?

**Comment by Ivshti** [comment]
@RSATom no. But you can override electron version passed to wcjs-prebuilt to something compatible of what you are using

**Comment by Ivshti** [comment]
If you remember, I had made separate electron runtime names `electron44` and `electron45` to fix this issue. Luigi's latest improvements wiped that out, but we can return it

**Comment by RSATom** [comment]
I think we need more generic solution

**Comment by RSATom** [comment]
btw, another solution could be trigger CI build by watching new releases on https://github.com/mafintosh/electron-prebuilt
But don't sure how to do it yet

**Comment by RSATom** [comment]
I think better solution will be remove direct dependency from `wcjs`. I did it already in https://github.com/RSATom/webgl-video-renderer, but it has a little bit changed api

**Comment by luigiplr** [comment]
@jaruba I... Might be. :neutral_face:

@RSATom should not be to hard to setup a ci task to "watch" the github releases of the electron repo.

**Comment by RSATom** [comment]
@luigiplr, yes, but excluding direct dependency is still good idea, since some time later we will have `wcjs` with `gstreamer`, and I want use same renderer for both `wcjs` and `wcjs-gs` (that's why I did `webgl-video-renderer`)

**Comment by luigiplr** [comment]
Ahhh cant wait for gStreamer variant of wcjs :dancer:

**Comment by RSATom** [comment]
It's very possible it will be only for linux...

**Comment by luigiplr** [comment]
> It's very possible it will be only for linux...

![Not Aumsed](https://wesleynicholls.files.wordpress.com/2014/06/obama_not_amused.gif)

**Comment by RSATom** [comment]
I just think libvlc is better alternative on Windows.
